proptest/src/strategy/unions.rs

```diff
@@ -95,8 +95,12 @@ impl<T: Strategy> Union<T> {
                 <= u64::from(u32::MAX),
             "Union weights overflow u32"
         );
-        let options =
-            options.into_iter().map(|(w, v)| (w, Arc::new(v))).collect();
+        let options: Vec<WA<T>> = options
+            .into_iter()
+            .filter(|(w, _)| w > &0)
```

Comment:
⛏️  `*w > 0` ready easier to my eyes

Reply:
72a9be32b7bce310c4fe3c6ab94f3ec37fa502b2